src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UploadContestedCaseDocumentsService.java

```diff
@@ -0,0 +1,113 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.ContestedUploadedDocumentData;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_CORRESPONDENCE_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_EVIDENCE_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_FR_FORM_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_TRIAL_BUNDLE_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONTESTED_UPLOADED_DOCUMENTS;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class UploadContestedCaseDocumentsService {
+
+    private static final String APPLICANT = "applicant";
+
+    private final ObjectMapper mapper;
+
+    public Map<String, Object> filterDocumentsToRelevantParty(Map<String, Object> caseData) throws Exception {
+
+        List<ContestedUploadedDocumentData> uploadedDocuments = getDocumentCollection(caseData, CONTESTED_UPLOADED_DOCUMENTS);
+
+        //we need to build ApplicantUploadedDocumentData, and add the "Applicant" marked doc a collection
+        List<ContestedUploadedDocumentData> correspondenceFiltered = uploadedDocuments.stream()
+            .filter(d -> d.getUploadedCaseDocument().getCaseDocumentParty() != null
+                && d.getUploadedCaseDocument().getCaseDocumentParty().equals(APPLICANT))
+            .filter(d -> d.getUploadedCaseDocument().getCaseDocumentType().equals("Letter from Applicant"))
+            .collect(Collectors.toList());
+        List<ContestedUploadedDocumentData> applicantCorrespondenceCollection = getDocumentCollection(caseData, APPLICANT_CORRESPONDENCE_COLLECTION);
+        applicantCorrespondenceCollection.addAll(correspondenceFiltered);
```

Comment:
Could maybe stick the above in its own separate method to separate out from having one mono method and make it more readable

Reply:
Yep, done this now for Applicant and Respondent document types :)